src/OnboardingSPA/store/selectors.js

```diff
@@ -206,15 +206,24 @@ export function getStepFromPath( state, path ) {
  * @return {object|null|false} Previous Step
  */
 export function getPreviousStep( state ) {
-	const currentStepIndex = findIndex( state.flow.steps.allSteps, {
+	let currentStepIndex = findIndex( state.flow.steps.allSteps, {
 		path: state.flow.steps.currentStep,
 	} );
+	const currentStep = state.flow.steps.allSteps[ currentStepIndex ];
+
 	if ( 0 === currentStepIndex ) {
 		return null; // current step is the first step
 	}
 	if ( -1 === currentStepIndex ) {
 		return false; // could not find index
 	}
+	// if current step is header menu and customDesign is false then skip colors and typography step
+	if (
+		! state.data.flowData.data.customDesign &&
+		!! currentStep.excludePreviousStepsFromRouter
```

Comment:
Do we need the double not operator? Instead can we use it without any operator for `excludePreviousStepsFromRouter` and `excludeNextStepsFromRouter` ?
Can we also change the doc comment to a generic one for this condition here and in `getNextStep`?

Reply:
We need the "!!" operator to check whether the property exists for the current step, If the property is not defined it will return false. Have updated the comments to more generic ones